zenodo_accessrequests/views/requests.py

```diff
@@ -156,7 +156,22 @@ def confirm(pid, record, template, **kwargs):
 
     # Confirm email address.
     if r.status != RequestStatus.EMAIL_VALIDATION:
-        abort(404)
+        if(str(r.status.value).lower() == "pending"):
```

Comment:
```suggestion
        if r.status == RequestStatus.PENDING:
```

- No need for parentheses around `if ...` statements :)
- It's better to compare with the enum value directly